models/User.js

```diff
@@ -128,6 +131,19 @@ module.exports = function(sequelize, DataTypes) {
   return sequelize.define(name, attributes, options);
 };
 
+const changeGlobalPermission = async function(admin, user, permission) {
+  if (!user || !admin || !admin.isAdmin()) {
+    return false;
+  }
+  user.setDataValue('globalPermission', permission);
+  await user.save();
+  return true;
+};
+
+const isAdmin = function() {
+  return this.globalPermission == 'write';
+};
```

Comment:
You could replace this with a more general `hasPermission` and replace all the permission checks with calls to this.